app/api/v2/handlers/plugins_api.py

```diff
@@ -0,0 +1,46 @@
+import aiohttp_apispec
+from aiohttp import web
+
+from app.api.v2.handlers.base_api import BaseApi
+from app.api.v2.managers.base_api_manager import BaseApiManager
+from app.api.v2.responses import JsonHttpNotFound
+from app.api.v2.schemas.base_schemas import BaseGetAllQuerySchema, BaseGetOneQuerySchema
+from app.objects.c_plugin import PluginSchema
+
+
+class PluginApi(BaseApi):
+    def __init__(self, services):
+        super().__init__(auth_svc=services['auth_svc'])
+        self._api_manager = BaseApiManager(data_svc=services['data_svc'])
+
+    def add_routes(self, app: web.Application):
+        router = app.router
+        router.add_get('/plugins', self.get_plugins)
+        router.add_get('/plugins/{plugin_name}', self.get_plugin_by_name)
+
+    @aiohttp_apispec.docs(tags=['plugins'])
+    @aiohttp_apispec.querystring_schema(BaseGetAllQuerySchema)
+    @aiohttp_apispec.response_schema(PluginSchema(many=True))
+    async def get_plugins(self, request: web.Request):
+        sort = request['querystring'].get('sort', 'name')
+        include = request['querystring'].get('include')
+        exclude = request['querystring'].get('exclude')
+
+        plugins = self._api_manager.get_objects_with_filters('plugins', sort=sort, include=include, exclude=exclude)
+        return web.json_response(plugins)
+
+    @aiohttp_apispec.docs(tags=['plugins'])
+    @aiohttp_apispec.querystring_schema(BaseGetOneQuerySchema)
+    @aiohttp_apispec.response_schema(PluginSchema)
+    async def get_plugin_by_name(self, request: web.Request):
+        plugin_name = request.match_info['plugin_name']
+        include = request['querystring'].get('include')
+        exclude = request['querystring'].get('exclude')
+
+        search = dict(name=plugin_name)
+
+        plugin = self._api_manager.get_object_with_filters('plugins', search=search, include=include, exclude=exclude)
```

Comment:
Does searching with access yield different results? If this was copied from planners -- they are a special case. See https://github.com/mitre/caldera/pull/2167/files#diff-8891c148b883a0802ff3991493a07472f78369ab22ecd01c8bfc2816acc59332R43 for an example.

Reply:
Thanks! I made the change, now only the plugins associated with the Red/Blue status of the user should be returned